pkg/operatorsource/purging.go

```diff
@@ -56,16 +55,6 @@ func (r *purgingReconciler) Reconcile(ctx context.Context, in *v1alpha1.Operator
 
 	r.datastore.RemoveOperatorSource(in.GetUID())
 
-	builder := &CatalogSourceConfigBuilder{}
-	csc := builder.WithTypeMeta().
-		WithNamespacedName(in.Namespace, in.Name).
-		CatalogSourceConfig()
-
-	if err = r.client.Delete(ctx, csc); err != nil && !k8s_errors.IsNotFound(err) {
```

Comment:
>- Do not delete CatalogSourceConfig object(s) during purge, instead let the CatalogSourceConfig(s) be updated during the reconciliation process.

I don't follow how the `CatalogSourceConfig` will get updated during the reconciliation process given it's phase will remain at `Succeeded`. Shouldn't you force an update so that we get the current state from Quay? Or are you saying that it will get updated during the next sync loop?

Reply:
The `CatalogSourceConfig` object will get updated during the 'Configuring' phase.

Reply:
Right this makes sense. Basically the Purging phase, rather than deleting the object, just resets the status and then sets the next phase to Initial. That way the configuring reconciler just updates the already existing object and it should work.

Reply:
Ah yes, sorry I missed the bit of existing code below. Make sense now.